src/main/java/gov/cabinetoffice/gap/adminbackend/services/SpotlightSubmissionService.java

```diff
@@ -20,4 +25,30 @@ public SpotlightSubmission getSpotlightSubmission(UUID spotlightSubmissionId) {
                 "A spotlight submission with id " + spotlightSubmissionId + " could not be found"));
     }
 
+    public void addSpotlightBatchToSpotlightSubmission(UUID spotlightSubmissionId, SpotlightBatch spotlightBatch) {
+        final SpotlightSubmission spotlightSubmission = getSpotlightSubmission(spotlightSubmissionId);
+
+        final List<SpotlightBatch> existingBatch = spotlightSubmission.getBatches();
+        existingBatch.add(spotlightBatch);
+        spotlightSubmission.setBatches(existingBatch);
+
+        spotlightSubmissionRepository.save(spotlightSubmission);
+    }
+
+    public List<SpotlightSubmission> getSubmissionsByBySchemeIdAndStatus(Integer schemeId,
+            SpotlightSubmissionStatus status) {
+        return spotlightSubmissionRepository.findByGrantSchemeIdAndStatus(schemeId, status.toString());
+    }
+
+    public long getCountBySchemeIdAndStatus(Integer schemeId, SpotlightSubmissionStatus status) {
+        return spotlightSubmissionRepository.countByGrantSchemeIdAndStatus(schemeId, status.toString());
+    }
+
+    public String getLastSubmissionDate(Integer schemeId, SpotlightSubmissionStatus status) {
+        final List<SpotlightSubmission> spotlightSubmissions = getSubmissionsByBySchemeIdAndStatus(schemeId, status);
+        return spotlightSubmissions.stream().map(SpotlightSubmission::getLastSendAttempt).max(Instant::compareTo)
+                .map(date -> date.atZone(ZoneOffset.UTC).format(DateTimeFormatter.ofPattern("dd MMMM yyyy")))
+                .orElse(null);
```

Comment:
Is this null handled/expected behaviour? if not it may be better to throw a custom exception here and handle appropriately.  What do you think?